### PR TITLE
Bg_Lotus OK

### PR DESCRIPTION
**Pull request by Zelllll**

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c [hunk 1]
```diff
@@ -19,22 +28,151 @@ const ActorInit Bg_Lotus_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/ovl_Bg_Lotus_0x80AD6760/func_80AD6A88.asm")
+void BgLotus_WaitToAppear(BgLotus* this, GlobalContext* globalCtx) {
+    if (this->timer > 0) {
+        this->timer--;
+    } else if ((this->dyna.actor.xzDistToPlayer > 100.0f) && (this->dyna.actor.projectedPos.z < 0.0f)) {
+        this->dyna.actor.draw = BgLotus_Draw;
+        func_800C6314(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        Actor_SetScale(&this->dyna.actor, 0.1f);
+        this->dyna.actor.world.pos.y =
+            (this->unk160 < this->dyna.actor.floorHeight) ? this->dyna.actor.floorHeight : this->unk160;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c:157]: Did you try CLAMP_MIN?

File: src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.h [hunk 2]
```diff
@@ -5,9 +5,15 @@
...
 struct BgLotus;
 
+typedef void (*BgLotusActionFunc)(struct BgLotus*, GlobalContext*);
+
 typedef struct BgLotus {
-    /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x28];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ BgLotusActionFunc actionFunc;
+    /* 0x160 */ f32 unk160;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.h:13]: This is a height.

File: src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.h [hunk 3]
```diff
@@ -5,9 +5,15 @@
...
+    /* 0x164 */ u8 unk164;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.h:14]: This is a boolean that determines whether or not ripple effects are spawned.

**Review by Kenix3 — COMMENTED** [review]
Code looks good to me, a couple small things.
Pending build test.

File: linker_scripts/code_script.txt [hunk 4]
```diff
@@ -5879,9 +5879,10 @@ SECTIONS
     ovl_Bg_Lotus : AT(RomLocation)
     {
         build/src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.o(.text)
-        build/asm/ovl_Bg_Lotus_data.o(.data)
+        build/src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.o(.data)
         build/src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.o(.rodata)
-        build/asm/ovl_Bg_Lotus_rodata.o(.rodata)
+        build/src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.o(.rodata)
```
Comment (Rozelette) [inline linker_scripts/code_script.txt:5884]: This seems incorrect. Why is `.rodata` repeated?

File: src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c [hunk 5]
```diff
@@ -19,22 +28,151 @@ const ActorInit Bg_Lotus_InitVars = {
...
+    BgLotus* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void BgLotus_SetScaleXZ(BgLotus* this) {
+    f32 scale;
+
+    if (this->dyna.actor.params == 0) {
+        scale = sin_rad(this->timer * 0.7853982f) * ((0.014f * ((f32)this->timer / 8)) + 0.01f);
```
Comment (Rozelette) [inline src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c:66]: Does still match if you do `scale = sin_rad(this->timer * 0.7853982f) * ((0.014f * (this->timer / 8.0f)) + 0.01f);`? (no cast)

**Review by Rozelette — APPROVED** [review]
LGTM except for a couple minor things

**Pull request closed without merging**